lib/resourcebuilder/batch.go

```diff
@@ -45,40 +44,51 @@ func (b *jobBuilder) Do(ctx context.Context) error {
 	if b.modifier != nil {
 		b.modifier(job)
 	}
-	_, updated, err := resourceapply.ApplyJob(b.client, job)
-	if err != nil {
+	if _, _, err := resourceapply.ApplyJob(b.client, job); err != nil {
 		return err
 	}
-	if updated && b.mode != InitializingMode {
-		return WaitForJobCompletion(ctx, b.client, job)
+	if b.mode != InitializingMode {
+		_, err := checkJobHealth(ctx, b.client, job)
+		return err
 	}
 	return nil
 }
 
 // WaitForJobCompletion waits for job to complete.
 func WaitForJobCompletion(ctx context.Context, client batchclientv1.JobsGetter, job *batchv1.Job) error {
```

Comment:
No longer used or needed?

Reply:
> No longer used or needed?

We still need it; discussed in the last paragraph of 100231e7.